WTRSimlib/src/main/java/simulation/launch/UnityLauncher.java

```diff
@@ -0,0 +1,41 @@
+package simulation.launch;
+
+import java.io.IOException;
+
+class UnityLauncher implements Runnable {
+
+    String unityVersion;
+    String unityProjectPath;
+    String unityDefaultScenePath;
+
+    UnityLauncher(String unityVersion, String unityProjectPath, 
+        String unityDefaultScenePath) {
+            
+        this.unityVersion = unityVersion;
+        this.unityProjectPath = unityProjectPath;
+        this.unityDefaultScenePath = unityDefaultScenePath;
+    }
+
+    @Override
+    public void run() {
+        System.out.println("Launching Unity Client... ");
+        try {
+            Runtime.getRuntime().exec("C:\\Program Files\\Unity\\Hub\\Editor\\" 
+                + unityVersion + "\\Editor\\Unity.exe -projectPath " 
+                + unityProjectPath + " -openfile " + unityProjectPath 
+                + unityDefaultScenePath); // Launch unity
+
+            Runtime.getRuntime().addShutdownHook(new Thread() { 
+                public void run() { 
```

Comment:
I can't see this without warning you that shutdown hooks don't always get called
*grits teeth* it's fine it's fine just leave it

Reply:
Don't worry, this is just a precaution. There are two things that can kill Unity. This shutdown thread and Unity itself :D